src/NUnitFramework/tests/Constraints/NUnitEqualityComparerTests.cs

```diff
@@ -150,6 +150,15 @@ public void IEquatableDifferentTypesSuccess_WhenExpectedImplementsIEquatable()
         }
 
         [Test]
+        public void IEquatableSuccess_WhenExplicitInterfaceImplementation()
+        {
+            IEquatableWithExplicitImplementation x = new IEquatableWithExplicitImplementation(1);
+            IEquatableWithExplicitImplementation y = new IEquatableWithExplicitImplementation(1);
```

Comment:
I might be overly cautious here and it may be unneeded, but I would like to see a second test for two `IEquatableWithExplicitImplementation` with different values to show the comparer.AreEqual returns false.